plugins/org.eclipse.emf.ecoretools.ale.core/src/org/eclipse/emf/ecoretools/ale/core/parser/internal/DatedDslSemantics.java

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Inria and Obeo.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Inria - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.emf.ecoretools.ale.core.parser.internal;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.eclipse.emf.ecore.EOperation;
+import org.eclipse.emf.ecoretools.ale.core.parser.visitor.ParseResult;
+import org.eclipse.emf.ecoretools.ale.implementation.Method;
+import org.eclipse.emf.ecoretools.ale.implementation.ModelUnit;
+
+/**
+ * The semantics of a DSL at some point in time.
+ * <p>
+ * An instance of this class is not updated as the underlying resource
+ * defining the semantics (e.g. a <i>.dsl</i> or a <i>.ale</i> file) changes.
+ */
+public class DatedDslSemantics implements DslSemantics {
```

Comment:
Maybe the name of this class is misleading
Date implies to have a date information in the class but there is not
 while the purpose of the class seems to have an immutable DslSemantic ?

Reply:
Renamed as `ImmutableDslSemantics`